File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        String[] splitLogLine = logLine.split(":");
        return splitLogLine[splitLogLine.length-1].trim();
    }

    public static String logLevel(String logLine) {
        String[] splitLogLine = logLine.split("]:");
        String infoLogLevel = splitLogLine[0].replace('[', ' ');
        return infoLogLevel.toLowerCase().trim();
    }

    public static String reformat(String logLine) {
        return String.format("%s (%s)", message(logLine), logLevel(logLine));
    }
}

File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

let message (logLine: string): string =
    logLine.Substring(logLine.IndexOf " ").Trim()

let logLevel(logLine: string): string =
    logLine.Substring(0, logLine.IndexOf(" ")).Replace("[","").Replace("]","").Replace(":","").ToLower()

let reformat(logLine: string): string =
    $"{(message logLine)} ({(logLevel logLine)})"  
